src/java/edu/ucsd/library/dams/util/Ezid.java

```diff
@@ -120,6 +146,24 @@ public static void validate( String damsXML ) throws DocumentException, EzidExce
         }
     }
 
+    /**
+     * Validate that DAMS XML contains a DOI.
+     * @param damsXML DAMS4 XML content
+     * @throws EzidException if the validation fails
+     * @returns DOI identifier
+    **/
+    public static String doi_validate( Document doc ) throws DocumentException, EzidException {
+        // pre-validate
+
+        String doi = doc.valueOf("/rdf:RDF/*/dams:note/dams:Note[(dams:displayLabel = 'DOI') and contains(rdf:value, 'http://doi.org/')]/rdf:value");
```

Comment:
Do we need a more generic way to detect the DOI as what's done in https://github.com/ucsdlib/damsrepo/pull/75/files#diff-82f0c8d4bd23c2b1704b3e0a2a6a37cdR127 or not?

Reply:
@lsitu - Which generic way that you prefer?  I can make make an update to use it. Thanks

Reply:
@VivianChu It seems like many records may have the DOI minted directly from the DOI site. So I am wondering whether we need
to extract the DOI from the preferred citation note like https://github.com/ucsdlib/damsrepo/pull/75/files#diff-82f0c8d4bd23c2b1704b3e0a2a6a37cdR127 /rdf:RDF/*/dams:note/dams:Note[(dams:type = 'preferred citation' or dams:type = 'identifier') and contains(rdf:value, 'http://dx.doi.org/')] or not. Just curios.

Reply:
@lsitu - We need to extract DOI for the update.

Reply:
@VivianChu I am just wondering that those DOI minted from the site directly might not have the DOI identifier. So if we want to support it more generally, we may need to extract it from the citation note. However, I am fine if we don't want to consider those cases for this update DOI support.

Reply:
@lsitu - DOI will be added to citation note when the Mint DOI is clicked.  It should be the same for identifier and citation note.  https://github.com/ucsdlib/damsrepo/blob/develop/src/java/edu/ucsd/library/dams/api/DAMSAPIServlet.java#L3467
https://github.com/ucsdlib/damsrepo/blob/develop/src/java/edu/ucsd/library/dams/api/DAMSAPIServlet.java#L3459

Reply:
@VivianChu The cases that I am talking about is just like https://library.ucsd.edu/dc/collection/bb2563476x (http://lib-hydratail-prod.ucsd.edu:8080/dams/api/objects/bb2563476x), we don't have the DOI Identifier note in this case.

Reply:
@lsitu - I thought all DOI object has DOI identifier note.  I'll update the code to extract doi from citation note.

Reply:
@lsitu - The code has been updated and deployed to staging.  It works fine. Could you review again?  Thanks

Reply:
@VivianChu 👍  I think it looks good and this will cover most of the cases except something like http://lib-hydratail-prod.ucsd.edu:8080/dams/api/objects/bb6793765b.

Reply:
@lsitu - The code has been updated and deployed staging.  It works fine.  For example - https://ezid.cdlib.org/id/doi:10.5072/FK26D5XN6H  Could you review again?  Thanks

Reply:
@VivianChu It looks good. My only concern is that DOI might not have a fixed length:
http://www.doi.org/doi_handbook/1_Introduction.html#1.6.3